lib/exception_handling.rb

```diff
@@ -54,8 +54,15 @@ def logger
       @logger or raise ArgumentError, "You must assign a value to #{name}.logger"
     end
 
+    Deprecation3_0 = ActiveSupport::Deprecation.new('3.0', 'exception_handling')
+
     def logger=(logger)
-      @logger = logger.is_a?(ContextualLogger) ? logger : ContextualLogger.new(logger)
+      @logger = if logger.is_a?(ContextualLogger::LoggerMixin)
+                  logger
+                else
+                  Deprecation3_0.deprecation_warning('implicit extend with ContextualLogger::LoggerMixin', 'extend your logger instance or include your into your logger class first')
```

Comment:
When a caller triggers this deprecation warning, they only show one caller back. So the private approach wound up pointing at `logger=` which was very confusing. So instead I'm doing the deprecation inline here.